### DISCUSSION TITLE
3 lines of getServerSideProps gives "This Serverless Function has timed out"

### DISCUSSION THREAD

**Discussion by naorye** [vercel/next.js#40226] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 20.6.0: Wed Jan 12 22:22:42 PST 2022; root:xnu-7195.141.19~2/RELEASE_X86_64
    Binaries:
      Node: 14.20.0
      npm: 6.14.17
      Yarn: 1.22.4
      pnpm: N/A
    Relevant packages:
      next: 12.2.6-canary.8
      eslint-config-next: 12.2.2
      react: 18.2.0
      react-dom: 18.2.0

### How are you deploying your application? (if relevant)

Vercel - hooby plan

### Describe the Bug

I have a page with the following getServerSideProps:

```
const Page: NextPage<MyProps> = (props) => (
  <span>hello</span>
);

function getAccessToken(
  request: IncomingMessage | NextApiRequest | express.Request
) {
  try {
    const cookies: Record<string, string | undefined> = nookies.get({
      req: request,
    });
    return cookies.accessToken;
  } catch (error) {
    console.log("Error retrieving accessToken cookie", error);
    return undefined;
  }
}

export function getServerSideProps(context: GetServerSidePropsContext) {
  const accessToken = getAccessToken(context.req);
  if (!accessToken) {
    return {
      redirect: {
        permanent: false,
        destination: "/login",
      },
    };
  }

  const props: MyProps = { accessToken };

  return { props };
}
```
When I run this page I get:
```
This Serverless Function has timed out.

Your connection is working correctly.

Vercel is working correctly.

```

### Expected Behavior

This page should work.

### Link to reproduction

Don't have, this is issue specific to my account

### To Reproduce

I believe this is some account limitation and I am looking for clarification or approval that this is the case.

**Comment by balazsorban44** [comment]
This does not sound like a Next.js bug, so moving to discussions, but could you provide a deployment URL, your repository/full reproduction, and any logs from the Vercel dashboard?

Could you clarify, if you cannot reproduce this locally or on a different account other than your own hobby account?

**Comment by icyJoseph** [comment]
What's the size of the cookies header? I imagine a sufficiently large cookie payload would make `nookie` and in turn `cookie` walk the cookie string for quite a while. It looks like it is O(n), with double `indexOf` calls, one to find the `=` and one to find the `;`, with the possibility to backtrack, but in the best case scenario, it has to walk the string forward looking for `=` and `;`.

Not sure if it'd be enough to timeout a serverless function, but it might be worth exploring the possibility.

**Comment by rikkster** [comment]
I have exactly the same issue with Vercel and getServerSideProps in Next.js. I'm racking my brain trying to find a solution, the case is very unusual. If I find a solution, I'll write it here.

**Reply by GabenGar** [reply]
The issue in the OP is pretty vague, so I am not sure how you can have it "exactly the same", since it can depend on a lot of things.

**Comment by rikkster** [comment]
After many hours and attempts, I found a solution. The problem was that Vercel, on the free plan, basically does not allow the use of getServerSideProps, throwing this error. You need to switch to the PRO plan (or use another hosting provider, not Vercel). Additionally, it's not enough to just switch Vercel to the PRO plan; you also need to create a vercel.json file in the project root, specifying the addresses of dynamic pages with SSR and extend the memory and timeout settings for them. I set it up for all pages at the top, and just in case, added two more important routes (they were the ones not working). This worked. I hope it saves someone time, as I myself spent a long time searching for a solution. An example of the contents of vercel.json is provided below:

> {
  "framework": "nextjs",
  "functions": {
    "src/pages/**/*": {
      "memory": 3008,
      "maxDuration": 20
    },
    "src/pages/all-therapists/index.tsx": {
      "memory": 3008,
      "maxDuration": 24
    },
    "src/pages/all-therapists/[diseaseUrl].tsx": {
      "memory": 3008,
      "maxDuration": 24
    }
  }
}

**Reply by GabenGar** [reply]
Vercel doesn't disallow the use of `getServerSideProps()`, it's your endpoints returning bloated responses, as evidenced by requirements to increase both memory size and timeout duration.

**Reply by rikkster** [reply]
You are absolutely right, this is just a stream of my thoughts 😅
And the solution has been found, I hope it will help people!
